pkg/gitops/GitOpsConfigService.go

```diff
@@ -0,0 +1,471 @@
+	clusterService cluster.ClusterService, envService cluster.EnvironmentService, versionService argocdServer.VersionService,
+	gitFactory *util.GitFactory) *GitOpsConfigServiceImpl {
+	return &GitOpsConfigServiceImpl{
+		logger:           Logger,
+		gitOpsRepository: gitOpsRepository,
+		K8sUtil:          K8sUtil,
+		aCDAuthConfig:    aCDAuthConfig,
+		clusterService:   clusterService,
+		envService:       envService,
+		versionService:   versionService,
+		gitFactory:       gitFactory,
+	}
+}
+func (impl *GitOpsConfigServiceImpl) CreateGitOpsConfig(request *GitOpsConfigDto) (*GitOpsConfigDto, error) {
+	impl.logger.Debugw("gitops create request", "req", request)
+	dbConnection := impl.gitOpsRepository.GetConnection()
+	tx, err := dbConnection.Begin()
+	if err != nil {
+		return nil, err
+	}
+	// Rollback tx on error.
+	defer tx.Rollback()
+
+	existingModel, err := impl.gitOpsRepository.GetGitOpsConfigActive()
+	if err != nil && err != pg.ErrNoRows {
+		impl.logger.Errorw("error in creating new gitops config", "error", err)
+		return nil, err
+	}
+	if existingModel != nil && existingModel.Id > 0 {
+		existingModel.Active = false
+		existingModel.UpdatedOn = time.Now()
+		existingModel.UpdatedBy = request.UserId
+		err = impl.gitOpsRepository.UpdateGitOpsConfig(existingModel, tx)
+		if err != nil {
+			impl.logger.Errorw("error in creating new gitops config", "error", err)
+			return nil, err
+		}
+	}
+	model := &repository.GitOpsConfig{
+		Provider:      strings.ToUpper(request.Provider),
+		Username:      request.Username,
+		Token:         request.Token,
+		GitHubOrgId:   request.GitHubOrgId,
+		GitLabGroupId: request.GitLabGroupId,
+		Host:          request.Host,
+		Active:        true,
+		AuditLog:      models.AuditLog{CreatedBy: request.UserId, CreatedOn: time.Now(), UpdatedOn: time.Now(), UpdatedBy: request.UserId},
+	}
+	model, err = impl.gitOpsRepository.CreateGitOpsConfig(model, tx)
+	if err != nil {
+		impl.logger.Errorw("error in saving gitops config", "data", model, "err", err)
+		err = &util.ApiError{
+			InternalMessage: "gitops config failed to create in db",
+			UserMessage:     "gitops config failed to create in db",
+		}
+		return nil, err
+	}
+
+	clusterBean, err := impl.clusterService.FindOne(cluster.ClusterName)
+	if err != nil {
+		return nil, err
+	}
+	cfg, err := impl.envService.GetClusterConfig(clusterBean)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := impl.K8sUtil.GetClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	secret, err := impl.K8sUtil.GetSecretFast(impl.aCDAuthConfig.ACDConfigMapNamespace, GitOpsSecretName, client)
+	statusError, _ := err.(*errors.StatusError)
+	if err != nil && statusError.Status().Code != http.StatusNotFound {
+		impl.logger.Errorw("secret not found", "err", err)
+		return nil, err
+	}
+	if secret == nil {
+		data := make(map[string][]byte)
+		data["username"] = []byte(request.Username)
```

Comment:
base64 encoding is not required here?

Reply:
for username or for both username and token ?

Reply:
> base64 encoding is not required here?

for username or for both username and token ?

Reply:
in secret all data should be base64 encoded, either we have to encode or library will encode

Reply:
> in secret all data should be base64 encoded, either we have to encode or library will encode

okay, making changes around this.